docs/extensions.md

```diff
@@ -166,6 +166,22 @@ response = httpcore.request(
 )
 ```
 
+### `"target"`
+
+The target that is used as the HTTP target instead of the URL path.
```

Comment:
```suggestion
The target that is used as [the HTTP target instead of the URL path](https://datatracker.ietf.org/doc/html/rfc2616#section-5.1.2).

This enables support constructing requests that would otherwise be unsupported. In particular...

* Forward proxy requests using an absolute URI.
* Tunneling proxy requests using `CONNECT` with hostname as the target.
* Server-wide `OPTIONS *` requests.
```